### ISSUE TITLE
Update awsx to use docker-build

### ISSUE THREAD

**Issue by mjeffryes** [pulumi/pulumi-awsx#1349]
docker-build is build is now the recommended way to build docker images, so we need to update the AWSX components to use it.

**Comment by danielrbradley** [comment]
Yes I think this is a great idea. This gives users much more composable blocks to build from.

Option 1: Include the credentials as an output of the registry resource in awsx - an object containing `registry`, `username` & `password` properties. These must be marked as secrets.

Option 2: Create an invoke which returns the same properties. This could live in either aws or awsx.

Perhaps we do both?

**Comment by blampe** [comment]
@flostadler to clarify, we aren't adding any new resources only changing the internals of what's currently available. The legacy docker provider is showing its age, and it would be good to prevent issues like e.g. https://github.com/pulumi/pulumi-docker/issues/967#issuecomment-2141929454 from biting the user.

I agree that auth helpers would also be a great cohesive addition.

**Comment by j-fulbright** [comment]
Any ideas on when this make it out into production? We'd like to be able to use docker this way instead of having to make config changes
